### PR TITLE
prov/efa: Split `efa_rdm_ep.inject_size`, add support for remaining EP size opts

### PR DESCRIPTION
**Pull request by darrylabbate**

### CODE REVIEW

File: prov/efa/src/rdm/efa_rdm_ep.h [hunk 1]
```diff
@@ -65,7 +65,14 @@ struct efa_rdm_ep {
 	size_t mtu_size;
 	size_t rx_iov_limit;
 	size_t tx_iov_limit;
-	size_t inject_size;
+	size_t max_msg_size;		/**< #FI_OPT_MAX_MSG_SIZE */
+	size_t max_tagged_size;		/**< #FI_OPT_MAX_TAGGED_SIZE */
+	size_t max_rma_size;		/**< #FI_OPT_MAX_RMA_SIZE */
+	size_t max_atomic_size;		/**< #FI_OPT_MAX_ATOMIC_SIZE */
+	size_t inject_msg_size;		/**< #FI_OPT_INJECT_MSG_SIZE */
```
Comment (shijin-aws) [inline prov/efa/src/rdm/efa_rdm_ep.h:72]: I am considering whether we can make it as an array with index as op code.

**Review by shijin-aws — APPROVED** [review]

File: prov/efa/src/rdm/efa_rdm_ep.h (hunk 1 above)
Comment (shijin-aws) [reply prov/efa/src/rdm/efa_rdm_ep.h:72]: This is not a blocking comment though

**Comment by shijin-aws** [comment]
bot:aws:retest

**Comment by darrylabbate** [comment]
Pretty sure MSVC is unhappy with `typeof`

**Comment by zachdworkin** [comment]
you will need to rebase after #10419 is merged to get Intel CI to pass

**Comment by darrylabbate** [comment]
> Pretty sure MSVC is unhappy ~with `typeof`~

Hmm it definitely was `typeof`, but even specifying `/std:clatest` (C23 `typeof` support) resulted in Windows build errors. Not worth wrestling with the Visual Studio project config.

**Comment by darrylabbate** [comment]
bot:aws:retest

**Comment by shijin-aws** [comment]
Need to rebase and resolve the merge conflict

**Review by shijin-aws — APPROVED** [review]

**Pull request merged**